src/components/PaginatedTable/useScrollBasedChunks.ts

```diff
@@ -84,7 +92,7 @@ export const useScrollBasedChunks = ({
     return React.useMemo(() => {
         // boolean array that represents active chunks
         const activeChunks = Array(chunksCount).fill(false);
-        for (let i = startChunk; i <= endChunk; i++) {
+        for (let i = startChunk; i <= Math.min(endChunk, chunksCount); i++) {
```

Comment:
`endChunk` already has value calculated based on `chunksCount`, because of this code in `calculateVisibleRange`:
```ts
const end = Math.min(
    Math.floor(visibleEnd / rowHeight / chunkSize) + overscanCount, 
    Math.max(chunksCount - 1, 0),
);
```